image/image_test.go

```diff
@@ -2019,6 +2019,38 @@ func (s *imageSuite) TestPrepareClassicModelSnapsButNoArchFails(c *C) {
 	c.Assert(err, ErrorMatches, "cannot have snaps for a classic image without an architecture in the model or from --arch")
 }
 
+func (s *imageSuite) TestPrepareClassicModelNoModelAssertion(c *C) {
+	preparedir := c.MkDir()
+	s.setupSnaps(c, map[string]string{
+		"pc":        "canonical",
+		"pc-kernel": "canonical",
+		"core":      "canonical",
+	}, "")
+
+	// prepare an image with no model assetion but classic set to true
+	// to ensure the GenericClassicModel is used without error
+	err := image.Prepare(&image.Options{
```

Comment:
It looks like the tooling store cannot be mocked in a way that Prepare() will pick it up. Actually the Prepare() tests seem to exercise only the error handling before a tooling store is created internally. For this reason, this should probably call `SetupSeed(s.tsto, ..)` passing the right set of arguments like other tests (also tweak the name of the test then)

Reply:
This test is designed to cover some code that's been added in image.Prepare so I'm not sure if calling SetupSeed really buys us anything. The purpose of this test is to check that calling image.Prepare with opts.Classic == true and no ModelAssertion specified doesn't result in an error like it previously did.

Reply:
I hope you don't mind, but I pushed a patch to your branch which add some required mocking and makes the test work.

Also since it's a classic system, the pc, pc-kernel snaps are not really used (or expected tbh) so I tweaked that to use something more realistic.